File: src/test/java/ru/tinkoff/gatling/kafka/javaapi/examples/ProducerSimulation.java
package ru.tinkoff.gatling.kafka.javaapi.examples;

import io.gatling.javaapi.core.ScenarioBuilder;
import io.gatling.javaapi.core.Simulation;
import org.apache.kafka.clients.producer.ProducerConfig;
import ru.tinkoff.gatling.kafka.javaapi.protocol.KafkaProtocolBuilder;

import java.util.Map;

import static io.gatling.javaapi.core.CoreDsl.*;
import static ru.tinkoff.gatling.kafka.javaapi.KafkaDsl.kafka;

public class ProducerSimulation extends Simulation {

    private final KafkaProtocolBuilder kafkaConsumerConf =
            kafka().topic("test.topic")
            .properties(Map.of(ProducerConfig.ACKS_CONFIG, "1"));

    private final ScenarioBuilder scn = scenario("Basic")
            .exec(kafka("BasicRequest").send("foo"))
            .exec(kafka("dld").send("true", 12.0));

}

File: src/test/kotlin/ru/tinkoff/gatling/kafka/javaapi/examples/ProducerSimulation.kt
package ru.tinkoff.gatling.kafka.javaapi.examples

import io.gatling.javaapi.core.CoreDsl.*
import io.gatling.javaapi.core.Simulation
import org.apache.kafka.clients.producer.ProducerConfig
import ru.tinkoff.gatling.kafka.javaapi.KafkaDsl.*
import java.time.Duration

class ProducerSimulation : Simulation() {

    private val kafkaConsumerConf = kafka().topic("test.topic")
        .properties(mapOf<String, Any>(ProducerConfig.ACKS_CONFIG to "1",
            ProducerConfig.BOOTSTRAP_SERVERS_CONFIG to "localhost:9092",
            ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG to "org.apache.kafka.common.serialization.StringSerializer",
            ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG to "org.apache.kafka.common.serialization.StringSerializer"
        ))

    private val scn = scenario("Basic")
        .exec(kafka("BasicRequest").send("foo"))
        .exec(kafka("dld").send("true", "12.0"))

    init {
        setUp(
            scn.injectOpen(atOnceUsers(1))
        )
            .protocols(kafkaConsumerConf)
            .maxDuration(Duration.ofSeconds(120))
    }

}